src/zjs_modules.c

```diff
@@ -254,6 +264,53 @@ static ZJS_DECL_FUNC(process_exit)
     exit(status);
 }
 #endif
+#ifdef ZJS_DYNAMIC_LOAD
+void zjs_modules_check_load_file(char *file)
+{
+    // No file waiting to load, just return
+    if (load_file == NULL) {
+        return;
+    }
+
+    zjs_stop_js();
```

Comment:
Why do you have to wait for the next event loop to do this stop_js? It would be simpler if you just did it from run_js and you wouldn't need the global `load_file` variable. Since there's a "stopJS()" function which does that immediately that part at least must be okay?

Reply:
Because load_file is set from the JS API runJS.  If you stop the JS from inside a JS API call the world explodes and it has no way back.  So to get around this I mark the load_file from the JS API, then when the main loop comes back around it sees the JS requested it load a new JS. The main loop can safely stop the JS and start the new JS. Long story short, the JS has to be started and stopped from the main loop for this to work.